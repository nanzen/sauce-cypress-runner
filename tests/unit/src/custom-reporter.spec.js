jest.mock('mkdirp');
jest.mock('fs');
const mkdirp = require('mkdirp');
const fs = require('fs');
const path = require('path');
const MochaJUnitReporter = require('../../../src/custom-reporter');

describe('Custom Reporter', function () {
  describe('.report', function () {
    it('calls flush on a spec file with proper paths', function () {
      const { report } = MochaJUnitReporter.prototype;
      const ctx = {};
      ctx._runner = {suite: { file: 'spec/root/folder/path/to/spec.js'}};
      ctx.flush = jest.fn(function () {});
      ctx._options = {
        specFolder: path.join(process.cwd(), 'spec', 'root', 'folder'),
        specRoot: path.join(process.cwd(), 'spec', 'root'),
      };
      report.call(ctx, 'a', 'b');
      expect(ctx.flush.mock.calls).toMatchSnapshot();
    });
  });
  describe('.writeXmlToDisk', function () {
    it('maintains relative paths correctly (addresses DEVX-273)', function () {
      const { writeXmlToDisk } = MochaJUnitReporter.prototype;
      const xml = `<?xml version="1.0" encoding="UTF-8"?>
            <items>
                <item>One</item>
                <item>Two</item>
                <item>Three</item>
            </items>
        `.trim();
      const filepath = '/path/to/[suite].xml';
      const filename = 'subdir-a/subdir-b/test.spec.js';
      mkdirp.mockImplementation(function () {});
      fs.writeFileSync.mockImplementation(function () {});
      writeXmlToDisk(xml, filepath, filename);
      expect(fs.writeFileSync.mock.calls).toEqual([
        ['/path/to/subdir-a/subdir-b/test.spec.js.xml', xml, 'utf-8'],
      ]);
    });
  });
});