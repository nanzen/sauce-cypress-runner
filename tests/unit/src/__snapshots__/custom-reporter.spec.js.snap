// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Reporter .report calls flush on a spec file with proper paths 1`] = `
Array [
  Array [
    "a",
    "folder/path/to/spec.js",
    "b",
  ],
]
`;
